src/atama/head/head.cpp

```diff
@@ -0,0 +1,576 @@
+              }
+              break;
+            case 1:
+              scan_pan_angle -= scan_speed;
+              if (scan_pan_angle < scan_right_limit) {
+                scan_position = (scan_position + 1) % 3;
+                scan_direction = 0;
+              }
+              break;
+          }
+
+          break;
+        }
+
+      case SCAN_VERTICAL:
+      case SCAN_HORIZONTAL:
+        {
+          if (init_scanning()) {
+            double value_change;
+            scan_speed = 0.8;
+
+            scan_pan_angle = get_pan_angle();
+            scan_tilt_angle = get_tilt_angle();
+
+            if (scan_mode == SCAN_VERTICAL) {
+              value_change = 15.0;
+            } else {
+              value_change = 30.0;
+            }
+
+            if (get_tilt_angle() < (scan_bottom_limit + scan_top_limit) / 2) {
+              scan_position = 0;
+              scan_tilt_angle -= value_change;
+            } else {
+              scan_position = 1;
+              scan_tilt_angle += value_change;
+            }
+          }
+
+          if (scan_mode == SCAN_VERTICAL) {
+            scan_pan_angle = (scan_left_limit + scan_right_limit) / 2;
+            switch (scan_position) {
+              case 0:
+                scan_tilt_angle += scan_speed;
+                if (scan_tilt_angle > scan_top_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+                break;
+              case 1:
+                scan_tilt_angle -= scan_speed;
+                if (scan_tilt_angle < scan_bottom_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+                break;
+            }
+          } else {
+            scan_tilt_angle = -60.0;
+            switch (scan_position) {
+              case 0:
+                scan_pan_angle += scan_speed;
+                if (scan_pan_angle > scan_left_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+
+                break;
+              case 1:
+                scan_pan_angle -= scan_speed;
+                if (scan_pan_angle < scan_right_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+
+                break;
+            }
+          }
+          break;
+        }
+    }
+
+    pan_angle = pan_center + keisan::clamp(scan_pan_angle, right_limit, left_limit);
+    tilt_angle = tilt_center + keisan::clamp(scan_tilt_angle, bottom_limit, top_limit);
+  }
+
+  // set calculation result
+  if (!joints.empty()) {
+    joints[0].set_position(pan_angle);
+    joints[1].set_position(tilt_angle);
+  }
+}
+
+void Head::scan_custom(int scan_type)
+{
+  if (!check_time_belief()) {
+    return;
+  }
+
+  function_id = scan_type;
+  scan(scan_type);
+  scan_process();
+}
+
+double Head::calculate_distance_from_pan_tilt(double pan, double tilt)
+{
+  double distance = 0.0;
+
+  for (int i = 6; i >= 0; i--) {
+    for (int j = 0; j <= i; j++) {
```

Comment:
i suggest to use `--i` and `++j` instead.

see these lines too
https://github.com/ichiro-its/atama/blob/fdd5eeceff0fc9eb8131e6839c22fd8620d3f4ca/src/atama/head/head.cpp#L364
https://github.com/ichiro-its/atama/blob/fdd5eeceff0fc9eb8131e6839c22fd8620d3f4ca/src/atama/head/head.cpp#L376-L377

Reply:
done